clang/lib/Sema/SemaChecking.cpp

```diff
@@ -12464,7 +12487,7 @@ static void AnalyzeImplicitConversions(
           << OrigE->getSourceRange() << T->isBooleanType()
           << FixItHint::CreateReplacement(UO->getBeginLoc(), "!");
 
-  if (const auto *BO = dyn_cast<BinaryOperator>(SourceExpr))
+  if (auto *BO = dyn_cast<BinaryOperator>(SourceExpr))
```

Comment:
Why do we lose `const` here?

Reply:
`AnalyzeImplicitConversions` takes a non-const `Expr *` and is called by `AnalyzeCommaOperator` from here. So it was this or a `const_cast` elsewhere.

Reply:
Ah, booo.  Ok then.